upup/models/cloudup/resources/addons/networking.flannel/k8s-1.12.yaml.template

```diff
@@ -86,37 +140,43 @@ data:
       }
     }
 ---
-kind: DaemonSet
 apiVersion: apps/v1
+kind: DaemonSet
 metadata:
   name: kube-flannel-ds
   namespace: kube-system
   labels:
-    k8s-app: flannel
+    tier: node
+    app: flannel
     role.kubernetes.io/networking: "1"
 spec:
   selector:
     matchLabels:
-      tier: node
```

Comment:
I think the selector field of a daemonset is immutable which would cause applying this manifest on an existing cluster to fail.

Reply:
Right as usual:
```
 The DaemonSet "kube-flannel-ds" is invalid: spec.selector: Invalid value:
v1.LabelSelector{MatchLabels:map[string]string{"app":"flannel"},
MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```